## Импорт класса

In [1]:
from email_preprocess import EmailPreprocess

## Создание экземпляра класса EmailPreprocess

In [2]:
preprocessor = EmailPreprocess(filename='data/emails.csv')

## Загрузка данных

In [3]:
data = preprocessor.load_data()
preprocessor.data.info()
preprocessor.data.sample(3)

Данные загружены
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5172 entries, 0 to 5171
Columns: 3002 entries, Email No. to Prediction
dtypes: int64(3001), object(1)
memory usage: 118.5+ MB


Email No.  the  to  ect  and  for  of   a  you  hou  ...  connevey  \
2801  Email 2802    4   4    7    0    3   2  31    2    3  ...         0   
3668  Email 3669    3   1    1    2    1   0  14    2    0  ...         0   
3893  Email 3894    1   3    2    0    0   0  15    2    0  ...         0   

      jay  valued  lay  infrastructure  military  allowing  ff  dry  \
2801    0       0    0               0         0         0   0    0   
3668    0       0    0               0         0         0   0    0   
3893    0       0    0               0         0         0   1    0   

      Prediction  
2801           0  
3668           0  
3893           1  

[3 rows x 3002 columns]

Выводы:
- в данных 3002 признака
- 3001 признак числовой, 1 признак - текстовый

## Проверка пропусков

In [4]:
preprocessor.check_missing_values()

Пропусков не обнаружено


## Разделение набора данных на признаки и метки

In [5]:
preprocessor.split_dataset()
preprocessor.X.shape, preprocessor.y.shape

Набор данных разделён на признаки и метки


((5172, 3000), (5172,))

## Очищение текстов

In [6]:
preprocessor.text_cleaning()
preprocessor.X.sample(3)

Текст писем очищен от лишних символов, пунктуации и стоп-слов


the  to  ect  and  for  of   a  you  hou  in  ...  enhancements  \
3652    5   6   11    3    1   4  42    4    5   4  ...             0   
1748    0   4    1    0    0   1   7    0    0   4  ...             0   
4154    0   5    2    0    2   0  55    0    0   4  ...             0   

      connevey  jay  valued  lay  infrastructure  military  allowing  ff  dry  
3652         0    0       0    0               0         0         0   3    0  
1748         0    0       0    0               0         0         0   2    0  
4154         0    0       0    0               0         0         0   0    0  

[3 rows x 3000 columns]

## Масштабирование признаков

In [7]:
preprocessor.feature_scaling()
preprocessor.X[:5]

Произведено масштабирование признаков


array([[-0.56544926, -0.64908256, -0.2938948 , ..., -0.0562853 ,
        -0.32904848, -0.07097072],
       [ 0.11575699,  0.71450797,  1.33733653, ..., -0.0562853 ,
         0.03067224, -0.07097072],
       [-0.56544926, -0.64908256, -0.2938948 , ..., -0.0562853 ,
        -0.32904848, -0.07097072],
       [-0.56544926, -0.12462466,  1.19549033, ..., -0.0562853 ,
        -0.32904848, -0.07097072],
       [ 0.03060621, -0.01973308,  0.84087482, ..., -0.0562853 ,
         0.03067224, -0.07097072]])

## Разделение на обучающий и тестовый наборы данных

In [8]:
preprocessor.train_test_split()
preprocessor.X_train.shape, preprocessor.X_test.shape, preprocessor.y_train.shape, preprocessor.y_test.shape

Данные разделены на обучающий и тестовый наборы


((4137, 3000), (1035, 3000), (4137,), (1035,))

## Выбор наиболее важных признаков

In [9]:
preprocessor.feature_selection(method='f_classif', k=10)
preprocessor.X_train.shape, preprocessor.X_test.shape

Отобраны наиболее значимые признаки


((4137, 10), (1035, 10))

## Полная предварительная обработка данных

In [10]:
preprocessor.preprocess()
preprocessor.X_train.shape, preprocessor.X_test.shape

Данные загружены
Пропусков не обнаружено
Набор данных разделён на признаки и метки
Текст писем очищен от лишних символов, пунктуации и стоп-слов
Произведено масштабирование признаков
Данные разделены на обучающий и тестовый наборы
Произведена предварительная обработка данных


((4137, 3000), (1035, 3000))

## Заключение:
- написан класс для предварительной обработки данных;
- проверена работоспособность методов данного класса.